### PR TITLE
[PUBDEV-6762] automate execution of testdir_hdfs tests

### PR DESCRIPTION
**Pull request by honzasterba**
step one: create an image and add the first test

### CODE REVIEW

File: docker/Jenkinsfile-build-hadoop-docker [hunk 1]
```diff
@@ -132,3 +131,63 @@ parallel(pipelineContext.getBuildConfig().getSupportedHadoopDistributions().coll
...
+                             usernamePassword(credentialsId: "${params.dockerRegistry}", usernameVariable: 'REGISTRY_USERNAME', passwordVariable: 'REGISTRY_PASSWORD')]) {
+                dir("${imageName}.${version}") {
+                    pipelineContext.getUtils().unstashFiles(this, DOCKER_STASH)
+                    sh """
+                                    cd docker
+
+                                    docker build \
+                                        -t ${REGISTRY_PREFIX}/${imageName}:${version} \
+                                        -f hadoop/slave/Dockerfile \
+                                        --build-arg PATH_PREFIX=hadoop/slave \
```
Comment (michalkurka) [inline docker/Jenkinsfile-build-hadoop-docker]: I think "slave" is a misleading name, could it be edge_node instead or similar?

File: docker/hadoop/slave/conf/core-site.xml [hunk 2]
```diff
@@ -0,0 +1,149 @@
+<!--Wed Dec  4 13:10:01 2019-->
```
Comment (michalkurka) [inline docker/hadoop/slave/conf/core-site.xml]: I don't think this file should be in this repository, it is better to automatically download a fresh copy of the hadoop conf directory a mount it in the docker container in every single run and thus make it fully dynamic.

**Comment by michalkurka** [comment]
@honzasterba I very much like this approach, the only thing I would suggest to change is to avoid including the Hadoop configuration in the git repo

File: docker/Jenkinsfile-build-hadoop-docker (hunk 1 above)
Comment (honzasterba) [reply docker/Jenkinsfile-build-hadoop-docker]: was not happy about the name too, will fix

File: docker/hadoop/slave/conf/core-site.xml (hunk 2 above)
Comment (honzasterba) [reply docker/hadoop/slave/conf/core-site.xml]: sounds good, will do

**Comment by honzasterba** [comment]
it works!
http://mr-0xc1:8080/view/H2O-3/job/h2o-hadoop-multinode-pipeline/

**Pull request merged**